### ISSUE TITLE
`response.assertAuthenticated()` doesn't work

### ISSUE THREAD

**Issue by os-nikita** [MasoniteFramework/masonite#646]
### Describe the bug

The testing framework does not recognise the authenticated user in the response.

### Steps to reproduce the bug

```python


ROUTES = [
    Route.get('/', "HomeController"),
]

from masonite.authentication import Auth
from masonite.controllers import Controller
from masonite.views import View

from app.models import User

class HomeController(Controller):
    def __call__(self, view: View, auth: Auth):
        user = User().create(
            {
                "name": "Jack Sparrow",
                "email": "js@pirat.es",
                "password": "doesn't matter",
            }
        )
        auth.attempt_by_id(user.id)
        return view.render('home')


class TestHomeController(TestCase):

    def test_something(self):
        self.actingAsGuest()
        response = self.get("/")
        response.assertAuthenticated()
```

### OS

macOS

### OS version

Monterey 12.3.1

### Masonite Version

4.12.1

**Comment by girardinsamuel** [comment]
I cannot reproduce the issue. It's working on my side 🤔 
Does it work for your when removing `self.actingAsGuest()` in the test ?

**Comment by os-nikita** [comment]
@girardinsamuel nope, also fails. Here's an even simpler replication. This is pretty much a base, skeletal Masonite installation, latest core (`4.14.3`) and orm (`2.13.0`).

```python

# HomeController.py

from masonite.controllers import Controller
from masonite.authentication import Auth
from masonite.views import View

from app.models import User


class HomeController(Controller):
    def __call__(self, view: View, auth: Auth):
        user = User().query().first()

        auth.attempt_by_id(user.id)

        return view.render('home')


# test_home_controller.py

from tests.TestCase import TestCase


class TestHomeController(TestCase):

    def test_something(self):
        response = self.get("/")
        response.assertAuthenticated()
```

Run with `python3 -m pytest`

**Comment by os-nikita** [comment]
Ah, I know why this is happening. This relates to the Auth singleton issue that I brought up a month or so ago on Discord.

I have a workaround for that issue in my subclassed AuthenticationProvider to get around that issue:

```python
from masonite.providers import AuthenticationProvider as MasoniteAuthenticationProvider


class AuthenticationProvider(MasoniteAuthenticationProvider):
    def boot(self):
        auth = self.application.make("auth")
        auth.set_user(None)
```

And that's causing this test to fail. If I remove it, it works, but the underlying Auth issue is there.

**Comment by girardinsamuel** [comment]
Yes. This should be fixed by #629 but has not been released because of the test failing...
